src/components/lab/canvas/molecules/CanvasOzone.js

```diff
@@ -10,40 +11,79 @@ import {
   CANVAS_MOLECULES_DISTANCE_BETWEEN_VERTICAL_ATOMS,
   NEGATIVE_CHARGE,
   POSITIVE_CHARGE,
+  CANVAS_MOLECULE_AREA_STATE,
 } from '../../../../config/constants';
 import CanvasBondContainer from './CanvasBondContainer';
 
-const CanvasOzone = ({ x }) => {
-  const oxygenAtomRadius = CANVAS_ATOM_DIMENSIONS[OXYGEN.size];
+const CanvasOzone = ({ x, shouldOscillate }) => {
+  const spectrum = useSelector(({ lab }) => lab.spectrum);
+  const moleculesOnCanvas = useSelector(({ lab }) => lab.moleculesOnCanvas);
 
-  const moleculeCenterPoints = {
-    topOxygenAtomCenterPoint: {
-      x: x + CANVAS_OZONE_ANGLED_ATOMS_X_ADJUSTMENT_FACTOR * 0.5,
-      y:
-        CANVAS_MOLECULE_AREA_Y_POSITION -
-        oxygenAtomRadius -
-        oxygenAtomRadius -
-        CANVAS_MOLECULES_DISTANCE_BETWEEN_VERTICAL_ATOMS,
-    },
-    middleOxygenAtomCenterPoint: {
-      x: x - CANVAS_OZONE_ANGLED_ATOMS_X_ADJUSTMENT_FACTOR * 0.5,
-      y: CANVAS_MOLECULE_AREA_Y_POSITION,
-    },
-    bottomOxygenAtomCenterPoint: {
-      x: x + CANVAS_OZONE_ANGLED_ATOMS_X_ADJUSTMENT_FACTOR * 0.5,
-      y:
-        CANVAS_MOLECULE_AREA_Y_POSITION +
-        oxygenAtomRadius +
-        oxygenAtomRadius +
-        CANVAS_MOLECULES_DISTANCE_BETWEEN_VERTICAL_ATOMS,
-    },
+  // constants to determine initial center points of atoms in this molecule when component mounts
+  // the 'initial center points' are also used in the useEffect hook below, to reset an oscillating molecule
+  const oxygenAtomRadius = CANVAS_ATOM_DIMENSIONS[OXYGEN.size];
+  const topOxygenAtomInitialCenterPoint = {
+    x: x + CANVAS_OZONE_ANGLED_ATOMS_X_ADJUSTMENT_FACTOR * 0.5,
+    y:
+      CANVAS_MOLECULE_AREA_Y_POSITION -
+      oxygenAtomRadius -
+      oxygenAtomRadius -
```

Comment:
`- oxygenAtomRadius*2`?

Reply:
I split them this way to make it a bit clearer, i.e. start with the initial y position, go up _this_ oxygen atom radius, then the `VERTICAL_DISTANCE`, then the radius of the oxygen atom above it. I should put the `VERTICAL_DISTANCE` in between, though (I had done this for a few atoms but not all).